src/cursor/cur_json.c

```diff
@@ -267,19 +267,21 @@ __wt_json_alloc_unpack(WT_SESSION_IMPL *session, const void *buffer, size_t size
   WT_CURSOR_JSON *json, bool iskey, va_list ap)
 {
     WT_CONFIG_ITEM *names;
-    size_t needed;
+    size_t bytes_allocated, needed;
```

Comment:
As coded, the call to `__wt_realloc_def` will update `bytes_allocated` (a stack variable), and json->value_size (or key_size) will never be non-zero.  I think what is needed is `size_t *bytes_allocatedp`, and set `bytes_allocatedp = &json->key_size` and pass `bytes_allocatedp` directly to `__wt_realloc_def`.  Let me know if you want to talk through it.

Reply:
> As coded, the call to `__wt_realloc_def` will update `bytes_allocated` (a stack variable), and json->value_size (or key_size) will never be non-zero. I think what is needed is `size_t *bytes_allocatedp`, and set `bytes_allocatedp = &json->key_size` and pass `bytes_allocatedp` directly to `__wt_realloc_def`. Let me know if you want to talk through it.
> 
> After you make this change, please investigate if you can change the assertion back to the way it was in the realloc function?

I tried passing in the pointer directly as you suggested but the Python test fails to run. It's not clear from the output what's going wrong.